apps/full-site-editing/full-site-editing-plugin/full-site-editing/class-full-site-editing.php

```diff
@@ -584,4 +585,29 @@ public function restrict_template_taxonomy_deletion( $term, $taxonomy ) { // php
 			wp_die( esc_html__( 'Template Types cannon be deleted.' ) );
 		}
 	}
+
+	/**
+	 * Removes wp admin menu items we don't want like Customize and Widgets.
+	 */
+	public function remove_wp_admin_menu_items() {
+		global $submenu;
+
+		// For safety.
+		if ( ! \A8C\FSE\is_full_site_editing_active() ) {
```

Comment:
In which is this check needed? As far as I can see, the `admin_menu` action is only added in constructor, which in turn is only loaded when FSE is active, so this seems redundant?

We are also hooking many other actions there that could mess up non-FSE sites - if there was a problem we'd likely hear about it by now.

Reply:
you're definitely right! Personally, I like leaving it because I don't think it could hurt anything.

Reply:
In my opinion it can confuse the readers of the code along the lines of - oh this is probably always executed so I need to make sure to add explicit bails in my new method too. And then this pattern can needlessly proliferate.

Also if there is an edge case in our loading strategy, I'd rather expose it as soon as possible, instead of hiding it. :)

Reply:
that is a good point. I'll avoid it in the future!